desktop-widgets/divepicturewidget.cpp

```diff
@@ -59,3 +59,16 @@ void DivePictureWidget::mousePressEvent(QMouseEvent *event)
 		QListView::mousePressEvent(event);
 	}
 }
+
+void DivePictureWidget::wheelEvent(QWheelEvent *event)
+{
+	if (event->modifiers() == Qt::ControlModifier) {
+		// Angle delta is given in eights of degrees. A typical mouse wheel
+		// tick is 15 degrees. Therefore, divide by 15*8=120, rounding away
+		// from 0.
```

Comment:
How the heck does that work for people with track pads?
I literally had to read your comment twice because I haven't used a mouse with a wheel in... I don't know... more than a decade?

Reply:
A quick test with a Thinkpad trackpad on Linux and this seems to be working. The only compliant I have is that I have to be on top of the zoom slider. I was first sure this doesn't work as I was trying to zoom in the thumbnail area, that seems logical to me...

Reply:
I agree - it would make sense to be able to do that on the thumbnail area - that should be an easy modification (famous last words)

Reply:
Note that in the thumbnail area you want the scroll wheel do scrolling of the pictures. Therefore this code is active only if pressing CTRL (`event->modifiers() == Qt::ControlModifier`). The mouse wheel on the slider is automatically done by Qt. See @sfuchs79's suggestion in #898.

Reply:
> How the heck does that work for people with track pads?

I (hopefully) improved the comment:

The whole point of the code is to work according to expectations of mouse wheel *and* track pad users. Mouse wheel users expect one click = one zoom level. Therefore divide by 120. But this would eat events < 120 generated by track pads, therefore round away from 0.

A more primitive way would be to always step +/-1 for any event. But then two quick clicks in one event would only step one level.

A more involved way would be actually keeping track of the deltas in a static variable, but in my quick testing the divide by 120 way was good enough. BTW: I didn't come up with all that - this is pretty much what the official QWheelEvent documentation says.

PS: We could avoid all that stuff if we simply could catch the event, remove the CTRL-modifier and pass it to the slider. I just don't know how that would be done in an idiomatic way.